### PR TITLE
fix: Deno.noColor should not be true when NO_COLOR is empty string

### PR DESCRIPTION
**Pull request by dsherret**
Closes https://github.com/denoland/deno/issues/21274

### CODE REVIEW

File: cli/tests/unit/tty_color_test.ts [hunk 1]
```diff
@@ -24,3 +24,32 @@ Deno.test(
...
+  },
+);
+
+Deno.test(
+  { permissions: { run: true, read: true } },
+  async function denoNoColorTrueEmptyVar() {
+    const { stdout } = await new Deno.Command(Deno.execPath(), {
+      args: ["eval", "console.log(Deno.noColor)"],
+      env: {
+        NO_COLOR: "1",
```
Comment (aapoalas) [inline cli/tests/unit/tty_color_test.ts:49]: **question:** Does `NO_COLOR=0` also turn on noColor? What's the usual way env flags work with regards to this?

Comment (dsherret) [reply cli/tests/unit/tty_color_test.ts:49]: According to this https://no-color.org/

> Command-line software which adds ANSI color to its output by default should check for a NO_COLOR environment variable that, when present and not an empty string (regardless of its value), prevents the addition of ANSI color.

We don't necessarily have to follow that though...

Comment (aapoalas) [reply cli/tests/unit/tty_color_test.ts:49]: This seems like an excellent "standard" to follow. Thank you!

**Review by aapoalas — APPROVED** [review]
Looks good to me!

**Pull request merged**